skore/src/skore/sklearn/_estimator/metrics_accessor.py

```diff
@@ -448,6 +455,238 @@ def _compute_metric_scores(
+        ...     classifier,
+        ...     X_train=X_train,
+        ...     y_train=y_train,
+        ...     X_test=X_test,
+        ...     y_test=y_test,
+        ... )
+
+        >>> report.metrics._predict_time()  # Predictions on the test data
+        {'predict_test': ...}
+
+        >>> report.metrics._predict_time(data_source="train")
+        {'predict_train': ...}
+
+        >>> report.metrics._predict_time(data_source="X_y", X=X_test, y=y_test)
+        {'predict_X_y_f41066b51130a331835e7d66417ff728': ...}
+        """
+        X, y, data_source_hash = self._get_X_y_and_data_source_hash(
+            data_source=data_source, X=X, y=y
+        )
+
+        def find_in_cache():
+            # Search for first matching key in cache
+            if data_source_hash is None:
+                for key in self._parent._cache:
+                    if key[-2:] == (data_source, "predict_time"):
+                        return self._parent._cache[key]
+            else:
+                for key in self._parent._cache:
+                    if key[-3:] == (data_source, data_source_hash, "predict_time"):
+                        return self._parent._cache[key]
+            return None
+
+        predict_time = find_in_cache()
+
+        if predict_time is None:
+            # If the time was not found in the cache, compute predictions
+            _get_cached_response_values(
+                cache=self._parent._cache,
+                estimator_hash=self._parent._hash,
+                estimator=self._parent.estimator_,
+                X=X,
+                response_method="predict",
+                data_source=data_source,
+                data_source_hash=data_source_hash,
+            )
+
+            predict_time = find_in_cache()
+
+            if predict_time is None:
+                Exception("This error should never be reached")
+
+        data_source_ = (
+            f"X_y_{data_source_hash}" if data_source == "X_y" else data_source
+        )
+        return {f"predict_{data_source_}": predict_time}
+
+    def timing(
+        self,
+        *,
+        kind: Literal["fit", "predict", "auto"] = "auto",
+        data_source: DataSource = "test",
+        X: Optional[ArrayLike] = None,
+        y: Optional[ArrayLike] = None,
+    ) -> dict:
+        """Report fit and prediction times, in seconds.
+
+        The times are reported in the form of a dict of the following shape:
+
+        - If kind="fit", a single entry with key "fit"
+        - If kind="predict", a single entry with key "predict_{data_source}"
+          (where data_source is "train", "test" or "X_y_{data_source_hash}")
+        - If kind="auto", two entries: a concatenation of the "fit" and "predict" cases.
+
+        Fit information might not be available, e.g. if the estimator was fitted outside
+        of the EstimatorReport. In this case, passing kind="fit" will result in an
+        error, and passing kind="auto" is equivalent to passing kind="predict".
```

Comment:
I'm fine with this one. Can you mention that `"predict"` will potentially request the computation of the predictions if not available in the local cache.